src/xtopcom/xvm/xsystem_contracts/src/xtable_reward_claiming_contract.cpp

```diff
@@ -95,8 +97,151 @@ void xtop_table_reward_claiming_contract::update_vote_reward_record(common::xacc
+            uint64_t lock_time{0};
+            base::xstream_t first_stream{xcontext_t::instance(), (uint8_t *)v.first.data(), static_cast<uint32_t>(v.first.size())};
+            first_stream >> duration;
+            first_stream >> lock_time;
+
+            base::xstream_t second_stream{xcontext_t::instance(), (uint8_t *)v.second.data(), static_cast<uint32_t>(v.second.size())};
+            second_stream >> vote_num;
+            xdbg("[xtop_table_reward_claiming_contract::recv_voter_dividend_reward] reward-claiming pledge info, table_id:%d, account:%s, vote_num:%d,duration:%d,lock_time:%d",
+                 table_id,
+                 account,
+                 vote_num,
+                 duration,
+                 lock_time);
+            vote_sum += vote_num;
+
+            std::cout << "account: " << account;
+            std::cout << ", m_timer_height: " << m_timer_height;
+            std::cout << ", lock_time: " << lock_time;
+            std::cout << ", duration: " << duration;
+            std::cout << ", vote_num: " << vote_num << std::endl;
+#ifdef PERIOD_MOCK
+            if (m_timer_height - lock_time >= duration / 60) {
+#else
+            if (m_timer_height - lock_time >= duration * 24 * 60 * 6) {
+#endif
+                std::cout << m_timer_height << " - " << lock_time << " >= " << duration * 24 * 60 * 6 << std::endl;
+                if (0 != duration) {  // if not calculated in XPROPERTY_EXPIRE_VOTE_TOKEN_KEY
+                    calc_expire_token += store::xaccount_context_t::get_top_by_vote(vote_num, duration);
+                }
+            } else {
+                unexpire_vote_num += vote_num;
+            }
+        }
+        auto valid_vote_sum = (stored_expire_token + calc_expire_token) / 1000000 + unexpire_vote_num;
+        std::cout << "account: " << account;
+        std::cout << ", stored_expire_token: " << stored_expire_token;
+        std::cout << ", calc_expire_token: " << calc_expire_token;
+        std::cout << ", unexpire_vote_num: " << unexpire_vote_num;
+        std::cout << ", valid_vote_sum: " << valid_vote_sum;
+        std::cout << ", vote_sum: " << vote_sum << std::endl;
+        xdbg(
+            "[xtop_table_reward_claiming_contract::recv_voter_dividend_reward] reward-claiming valid vote info,table_id:%d,"
+            "account:%s,stored_expire_token:%d,calc_expire_token:%d,unexpire_vote_num:%d,valid_vote_sum:%d,vote_sum:%d",
+            table_id,
+            account,
+            stored_expire_token,
+            calc_expire_token,
+            unexpire_vote_num,
+            valid_vote_sum,
+            vote_sum);
+
+        for (auto & v : votes_table) {
+            auto adv = v.first;
+            auto valid_vote = (v.second * valid_vote_sum) / vote_sum;
+            std::cout << "adv: " << adv;
+            std::cout << ", vote: " << v.second;
+            std::cout << ", valid_vote: " << valid_vote << std::endl;
+            votes_table[adv] = valid_vote;
```

Comment:
you are iterating the votes_table, thus update `votes_table[adv] = valid_vote;` is better to be replaced as `top::get<uint64_t>(v) = valid_vote` or `v.second = valid_vote`, both should be same as `votes_table[adv] = valid_vote`